### ISSUE TITLE
Jest with jest-transform still giving (0 , _sprinkles.s) is not a function error

### ISSUE THREAD

**Issue by Nvveen** [vanilla-extract-css/vanilla-extract#962]
### Describe the bug

According to [Test Environment](https://vanilla-extract.style/documentation/test-environments/), adding the transform to the Jest options, should solve this issue. There was a change since #731  so I'm not sure what I'm missing. My setup is a bit big, since I use Nx (and in my real project a lot more), but I managed to reproduce it with a sort-of minimal example.

![image](https://user-images.githubusercontent.com/486978/211430780-8c028184-e478-4ead-a38b-ddf9d4308e88.png)

### Reproduction

https://github.com/Nvveen/nx-sprinkles-repro

### System Info

```shell
System:
    OS: Linux 5.10 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700H
    Memory: 9.98 GB / 15.48 GB
    Container: Yes
    Shell: 5.7.1 - /usr/bin/zsh
  Binaries:
    Node: 16.19.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.19.3 - /usr/local/bin/npm
  npmPackages:
    @vanilla-extract/css: ^1.9.2 => 1.9.2 
    @vanilla-extract/jest-transform: ^1.0.1 => 1.0.1 
    @vanilla-extract/sprinkles: ^1.5.1 => 1.5.1 
    @vanilla-extract/vite-plugin: ^3.7.0 => 3.7.0 
    vite: ^4.0.1 => 4.0.4
```

### Used Package Manager

pnpm

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by rothsandro** [comment]
The Nx Jest Resolver uses `identitiy-obj-proxy` to resolve CSS files, see [`resolver.ts`](https://github.com/nrwl/nx/blob/5970246b51c17428b8d9f04123b06f780ac824bc/packages/jest/plugins/resolver.ts#L43-L51).

**Comment by asherccohen** [comment]
anyone got this to work?

**Comment by jd-oconnor** [comment]
> anyone got this to work?

This fixed my jest transform issue
https://bjgrosse.github.io/techblog/2023/01/04/vanilla-extract-and-jest-style-mocks.html

**Comment by asherccohen** [comment]
Hi Jake, we chatted on discord and came to the conclusion that NX is not ready for vanilla-extract.

Looking for assistance to find the right config, all my tests are broken 💔

**Comment by asherccohen** [comment]
See https://github.com/nrwl/nx/issues/12653

**Comment by mattcompiles** [comment]
Closed by #982
